src/coreclr/src/pal/src/locale/utf8.cpp

```diff
@@ -1586,13 +1586,8 @@ class UTF8Encoding
                         charCount--;
                     }
                 }
-                else {
-                    // 2 byte encoding
```

Comment:
Removing this comment does not make the code easier to understand. The extra else and scope are here on purpose to make the code easier to understand.